target_verifier.go

```diff
@@ -0,0 +1,54 @@
+package ghostferry
+
+import (
+	"fmt"
+
+	sql "github.com/Shopify/ghostferry/sqlwrapper"
+	"github.com/sirupsen/logrus"
+)
+
+type TargetVerifier struct {
+	logger         *logrus.Entry
+	DB             *sql.DB
+	BinlogStreamer *BinlogStreamer
+	StateTracker   *StateTracker
+}
+
+func NewTargetVerifier(targetDB *sql.DB, stateTracker *StateTracker, binlogStreamer *BinlogStreamer) (*TargetVerifier, error) {
```

Comment:
Traditionally I've used `Ferry` as a sort of factory, so you don't have to pass it all these variables (or worse, pass in the wrong variables). It's also somewhat natural to think of Ferry to be a factory for all the components of Ghostferry. Should this component not follow that convention? Do we not like the other convention?

Reply:
Do we want to keep that convention? It also makes some implicit decisions about what we want to use for the struct members that are built there. I can move it if so, but I thought I would take this opportunity to introduce a different pattern typically seen in go projects.

Reply:
Will keep as-is for now, and we can move it into the Ferry later if we decide we want to keep that pattern 👍